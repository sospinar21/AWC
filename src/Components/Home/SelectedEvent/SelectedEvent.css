.selected-event {
  background-size: cover;
  background-position-x: center;
  background-position-y: center;
  background-repeat: no-repeat;
  background-repeat: no-repeat;
  position: relative;
  height: 320px;
  width: 100%;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-shadow: 0px 0px 3px lightgrey;
  box-shadow: 0px 0px 3px lightgrey;
  background-image: url(http://www.semo.edu/images/dance-modernBanner.jpg); }

.event-desc {
  background-color: rgba(0, 0, 0, 0.74);
  position: absolute;
  bottom: 20px;
  padding: 23px;
  border-radius: 3px;
  width: 60%; }
  .event-desc h1 {
    color: #b62d4c;
    text-shadow: 1px 1px 1px black; }

.flex {
  display: flex; }

.event-description {
  padding: 0 10px;
  margin: 11px auto;
  height: 100px;
  overflow: scroll; }
  .event-description p {
    color: white; }
