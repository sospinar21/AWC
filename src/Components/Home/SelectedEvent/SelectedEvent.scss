.selected-event {
  background-size: cover;
  background-position-x: center;
  background-position-y: center;
  background-repeat: no-repeat;
  background-repeat: no-repeat;
  position: relative;
  height: 250px;
  width: 100%;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-shadow: 0px 0px 3px lightgrey;
  box-shadow: 0px 0px 3px lightgrey;
  background-image: url(http://www.marylandyouthballet.org/wp-content/uploads/2012/04/Lilly-Swan-Lake-Profile.jpg);
}


.flex{
  width: 100%;
  padding: 7px 10px;
  overflow-y: scroll;
  margin-bottom: 20px;

  h1 {
    font-size: 25px;
    color: #262834;
    text-shadow: 1px 1px 1px #f7f7f7;
  }
}


.event-description{
  padding: 0 10px;
  margin: 11px auto;
  overflow: scroll;

  p{
    font-size: 13px;
    line-height: 1.5;
    color: black;
    background-color: #ffffffa3;
  }
}