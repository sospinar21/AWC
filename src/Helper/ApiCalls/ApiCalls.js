import React, {Component} from 'react';

class ApiCalls extends Component {
  constructor(){
    super()
  }

  fetchEvents = async () => {
    const url = `https://api.awc.dance/events?city=vermont`;
    const response = await fetch(url)
    const data= await response.json();
    return data;
  } 

  fetchStudios = () => {
    const url = `https://api.awc.dance/?city=80202`;
    return fetch(url, {
      method: 'GET',
      headers: {
        'Content-type': 'application/json',
      }
    }).then( response => response.json() )
      .then( data => data )   
  }

  fetchVideos= async () => {
    var url = 'https://api.awc.dance/youtubevideos'
    const response = await fetch(url)
    const data = await response.json()
    return data
  }
}

export default ApiCalls;