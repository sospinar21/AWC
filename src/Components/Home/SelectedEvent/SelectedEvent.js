import React, { Component } from 'react';
import { connect } from 'react-redux';
import './SelectedEvent.css';
import PropTypes from 'prop-types';

export class SelectedEvent extends Component {

  backgroundImage = () => {
    if (this.props.selectedEvent.logo){
      const selected = this.props.selectedEvent.logo.url;
      const selectedEvent = document.querySelector('.selected-event')
      selectedEvent.style.backgroundImage = `url(${selected})`
    }
  }

render () {
    const selected = this.props.selectedEvent
    if (selected) {
      return (
        <div className='flex'>
          {this.backgroundImage()}
          <div className='event-desc'>
            <h1> {selected.name}</h1>
            <div className='event-description'>
              {/* <h2>{selected.date.local}</h2> */}
              <p>{selected.description}</p>
            </div>
          </div>
        </div>
      ); 
    } else {
      return (
        <div className='loading'>
         Loading...
        </div>
      );
    }
  }
}

export const mapStateToProps = (state) => {
  return ({
    selectedEvent: state.selectedEvent
  });
};

export default connect(mapStateToProps)(SelectedEvent);