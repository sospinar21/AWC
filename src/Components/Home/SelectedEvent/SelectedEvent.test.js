import {SelectedEvent, mapStateToProps} from '../SelectedEvent/SelectedEvent';
import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import ApiCalls from '../../../Helper/ApiCalls/ApiCalls';

describe('selectedEvent', () => {
  let selectedEvent;
  let mockProps;
  const api = new ApiCalls();

  beforeEach(() => {
    mockProps = {
      selectedEvent: {name: 'Workshop'},
    };

    selectedEvent = shallow(<SelectedEvent {...mockProps}/>, { disableLifecycleMethods: true });

  });

  it('matches snapshot', () => {
    let selectedEvent = shallow(<SelectedEvent {...mockProps}/>, { disableLifecycleMethods: true });
    expect(selectedEvent).toMatchSnapshot();
  });

  
  describe('mapStateToProps', () => {

    let mappedProps

    it('returns an object with the user', () => {

      const mockState = {
        selectedEvent: {name: 'workshop'}
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockState);
      
    })
  })

});