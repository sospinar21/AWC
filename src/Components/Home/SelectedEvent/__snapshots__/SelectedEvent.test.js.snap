// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectedEvent matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectedEvent
    selectedEvent={
        Object {
            "name": "Workshop",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="event-desc"
>
        <h1>
                 
                Workshop
        </h1>
        <div
                className="event-description"
        >
                <p />
        </div>
</div>,
      "className": "flex",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
             
            Workshop
</h1>,
          <div
            className="event-description"
>
            <p />
</div>,
        ],
        "className": "event-desc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "Workshop",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Workshop",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p />,
            "className": "event-description",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="event-desc"
>
          <h1>
                     
                    Workshop
          </h1>
          <div
                    className="event-description"
          >
                    <p />
          </div>
</div>,
        "className": "flex",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
               
              Workshop
</h1>,
            <div
              className="event-description"
>
              <p />
</div>,
          ],
          "className": "event-desc",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "Workshop",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Workshop",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p />,
              "className": "event-description",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
